daemon/stores/workspaces.py

```diff
@@ -53,10 +53,62 @@ def update(
         return id
 
     @BaseStore.dump
-    def delete(self, id: DaemonID, **kwargs):
+    def delete(
+        self,
+        id: DaemonID,
+        container: bool,
+        network: bool,
+        files: bool,
+        everything: bool,
+        **kwargs,
+    ):
+        deleted_entities = []
+        if everything:
+            container = True
+            network = True
+            files = True
+
+        if id not in self:
+            raise KeyError(f'{colored(str(id), "cyan")} not found in store.')
+
+        if container:
+            container_id = self[id].metadata.container_id
+            if not container_id:
+                raise ValueError(
+                    f'There is no container to kill for store {colored(str(id), "cyan")}'
+                )
+
+            Dockerizer.rm_container(container_id)
+            self._logger.success(
+                f'{colored(container_id, "cyan")} is killed and removed from the store.'
+            )
+            self[id].metadata.container_id = None
+            deleted_entities.append(container_id)
+
+        # TODO: add network and file deletion
+        if everything:
+            Dockerizer.rm_image(id=self[id].metadata.image_id)
```

Comment:
Would need to check if container is deleted here.

Reply:
The container gets deleted above? If we reach this point it should always have been deleted. In case of errors it should have thrown before. So I think it should be safe here to delete the image and the network (deleting the network we may actually want to move out of here though as it can be done separately now).

We need to extend the delete logic anyway as also file deletion is missing here and flows etc are also not stopped/deleted